internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -229,9 +230,108 @@ func (r *Reconciler) reconcileControlPlane(ctx context.Context, s *scope.Scope)
 		return errors.Wrapf(err, "failed to update %s", tlog.KObj{Obj: s.Desired.ControlPlane.Object})
 	}
 
+	// If the ControlPlane has defined a current or desired MachineHealthCheck attempt to reconcile it.
+	if s.Desired.ControlPlane.MachineHealthCheck != nil || s.Current.ControlPlane.MachineHealthCheck != nil {
+		// Set the ControlPlane Object and the Cluster as owners for the MachineHealthCheck if a desired MachineHealthCheck exists.
+		if s.Desired.ControlPlane.MachineHealthCheck != nil {
+			s.Desired.ControlPlane.MachineHealthCheck.SetOwnerReferences([]metav1.OwnerReference{
+				*ownerReferenceTo(s.Desired.ControlPlane.Object),
+			})
+		}
+
+		// Reconcile the current and desired state of the MachineHealthCheck.
+		if err := r.reconcileMachineHealthCheck(ctx, s.Current.ControlPlane.MachineHealthCheck, s.Desired.ControlPlane.MachineHealthCheck); err != nil {
+			return errors.Wrapf(err, "failed to update %s", tlog.KObj{Obj: s.Desired.ControlPlane.MachineHealthCheck})
+		}
+	}
+	return nil
+}
+
+// reconcileMachineHealthCheck creates, updates, deletes or leaves untouched a MachineHealthCheck depending on the difference between the
+// current state and the desired state.
+func (r *Reconciler) reconcileMachineHealthCheck(ctx context.Context, current, desired *clusterv1.MachineHealthCheck) error {
+	log := tlog.LoggerFrom(ctx)
+
+	// If a current MachineHealthCheck doesn't exist but there is a desired MachineHealthCheck attempt to create.
+	if current == nil && desired != nil {
+		// First ensure the ownerReferences are valid.
+		refs := []metav1.OwnerReference{}
+		for _, ref := range desired.OwnerReferences {
+			currentRef := ref
+			ownerRef, err := r.resolveOwnerReferenceIfIncomplete(ctx, desired.Namespace, &currentRef)
+			if err != nil {
+				return errors.Wrapf(err, "failed to resolve owner reference %v for  %s", ref, tlog.KObj{Obj: desired})
+			}
+			refs = append(refs, *ownerRef)
+		}
+		desired.OwnerReferences = refs
+
+		log.Infof("Creating %s", tlog.KObj{Obj: desired})
+		if err := r.Client.Create(ctx, desired); err != nil {
+			return errors.Wrapf(err, "failed to create %s", tlog.KObj{Obj: desired})
+		}
+		r.recorder.Eventf(desired, corev1.EventTypeNormal, updateEventReason, "Created %q", tlog.KObj{Obj: desired})
+		return nil
+	}
+
+	// If a current MachineHealthCheck exists but there is no desired MachineHealthCheck attempt to delete.
+	if current != nil && desired == nil {
+		log.Infof("Deleting %s", tlog.KObj{Obj: current})
+		if err := r.Client.Delete(ctx, current); err != nil {
```

Comment:
Should we? That would end up with us logging and returning as if it was deleted. This is likely an edge case, and I guess it could impact the deleteMachineDeployment code as it stands. WDYT?

Reply:
I think essentially our goal here is that the MHC is gone in the end. We don't really care about how it was deleted.

If we fail here with `not found` the reconcile will return an error and the next one won't really care about it too :)

It's absolutely an edge case, but I think as we don't care about the error of the MHC is already gone when we try to delete it, we can also just directly ignore the error.